sklearn/preprocessing/_encoders.py

```diff
@@ -581,7 +582,7 @@ def get_feature_names(self, input_features=None):
         for i in range(len(cats)):
             names = [
                 input_features[i] + '_' + str(t) for t in cats[i]]
-            if self.drop is not None:
+            if (self.drop is not None) and (self.drop_idx_[i] is not None):
```

Comment:
I would also use only the fitted attribute `drop_idx_`

Reply:
See comment above.

Reply:
I think that we can get in a similar situation than above (I would need to find the corner case) but for sure it would be much cleaner to keep using fitted attributes in `transform`, `inverse_transform` and `get_feature_names` which will get information computed at `fit`.